### PR TITLE
Get width and height from JavaScript evaluation.

### LINKED ISSUE #29: :resize-to-contents seems to have no effect
I have the following settings in my config:

```
                            :resize-to-contents {:height? true :width? true}
```

yet the screenshots that exceed one page in size are all cutoff in the vertical direction.

### PR DESCRIPTION
**Pull request by jleonard-r7**
Fixes #29

### CODE REVIEW

**Comment by oliyh** [comment]
Hi,

Thanks for this! I would like to keep `kamera.core` agnostic of devcards, so this could better be exposed as a key in the `:resize-to-contents` options map (called `:dom-selector` or something, defaulting to `body` in core but overridden to the devcards selector in `kamera.devcards`). This would ensure users could retain the same behaviour if they wished as well.

**Comment by oliyh** [comment]
This is also something I would like to test on my projects to see what difference it would make

**Comment by jleonard-r7** [comment]
Makes sense. Ok, I will add the `:dom-selector` change to this PR and update accordingly.

**Comment by oliyh** [comment]
Sorry I don't think I made myself clear, I think the `:dom-selector` should be inside the `:resize-to-contents` map, e.g. `{:resize-to-contents {:width? true :height? true :dom-selector "body"}}` as they all relate to the same feature (keeps the fn signatures cleaner too)

**Comment by jleonard-r7** [comment]
Ah I agree. That makes sense. Will update later.

File: src/clj/kamera/devcards.clj [hunk 1]
```diff
@@ -100,6 +100,7 @@
      (let [target-urls (find-test-urls session)
            targets (map (fn [target-url]
                           {:url (str devcards-url target-url)
+                           {:resize-to-contents {:dom-selector "#com-rigsomelight-devcards-main"}}
```
Comment (oliyh) [inline src/clj/kamera/devcards.clj]: This looks like it's got an extra pair of braces? I also just realised circle wasn't building PRs, I've enabled that now

Comment (jleonard-r7) [reply src/clj/kamera/devcards.clj]: Where do you see an extra pair of braces?

File: src/clj/kamera/core.clj [hunk 2]
```diff
@@ -265,7 +270,8 @@
                                                    ;; see element-exists?
                          :assert?              true ;; runs a clojure.test assert on the expected/actual when true, makes no assertions when false
                          :resize-to-contents   {:height? true
-                                                :width? false}}
+                                                :width? false
+                                                :dom-selector "body"}}
```
Comment (jleonard-r7) [inline src/clj/kamera/core.clj:274]: @oliyh `:resize-to-contents` is a map, thus the inner pair of braces and the requirement for a `deep-merge` below.

File: src/clj/kamera/devcards.clj (hunk 1 above)
Comment (oliyh) [reply src/clj/kamera/devcards.clj]: `:resize-to-contents` should be a sibling of `:url`, i.e. the options map should be 
```
:url "..."
:resize-to-contents {:dom-selector "..."}
```

Or am I just reading totally wrong? (I'm on my phone)

Comment (jleonard-r7) [reply src/clj/kamera/devcards.clj]: Oh, I’m sorry. You are referring to the outer braces. Yes, those are wrong.

**Comment by jleonard-r7** [comment]
Is this ready to merge now?

**Comment by oliyh** [comment]
Hi,

Thanks for the updates. I am just going to try this build with my existing projects to see what happens as I am curious about how the javascript size and devcards selector will behave, and also to see what existing users can expect when they upgrade. I'll try to do this tomorrow.

Cheers

**Comment by jleonard-r7** [comment]
Ok, with default settings, the only change they should see would be that the bug is actually fixed; i.e., the screenshots will be sized to content in the _vertical_ direction. That will likely break tests where their content exceeded the vertical extent of one page (but that's most likely a good thing). They will either need to update their reference images or change the config to disable sizing to content in the vertical direction.

Everything else should be as expected / normal / least surprise.

**Pull request merged**